spec/rake_spec.rb

```diff
@@ -10,17 +10,108 @@
   end
 
   describe 'readyset' do
+    describe 'create_cache' do
+      context 'when given a query ID as an argument' do
+        it 'creates a cache from the given query' do
+          query = build_and_execute_proxied_query(:proxied_query)
+          build_and_execute_proxied_query(:proxied_query_2)
+
+          Rake::Task['readyset:create_cache'].execute(Rake::TaskArguments.new([:id], [query.id]))
+
+          caches = Readyset::Query::CachedQuery.all
+          expect(caches).to eq([build(:cached_query)])
+        end
+      end
+
+      context 'when given no arguments' do
+        it 'prints an error message' do
+          expect { Rake::Task['readyset:create_cache'].execute }.
+            to output("A query ID must be passed to this task\n").to_stdout
+        end
+      end
+    end
+
+    describe 'create_cache_always' do
+      context 'when given a query ID as an argument' do
+        it 'creates a cache from the given query with `always` set to true' do
+          query = build_and_execute_proxied_query(:proxied_query)
+          build_and_execute_proxied_query(:proxied_query_2)
+
+          Rake::Task['readyset:create_cache_always'].
+            execute(Rake::TaskArguments.new([:id], [query.id]))
+
+          caches = Readyset::Query::CachedQuery.all
+          expect(caches).to eq([build(:cached_query, always: true)])
+        end
+      end
+
+      context 'when given no arguments' do
+        it 'prints an error message' do
+          expect { Rake::Task['readyset:create_cache_always'].execute }.
+            to output("A query ID must be passed to this task\n").to_stdout
+        end
+      end
+    end
+
     describe 'caches' do
+      it 'prints a table with the caches that currently exist on ReadySet' do
+        build_and_create_cache(:cached_query)
+
+        expected_message = <<~TABLE
+          +--------------------+--------------------+---------------------------------+--------+-------+
+          | id                 | name               | text                            | always | count |
+          +--------------------+--------------------+---------------------------------+--------+-------+
+          | q_4f3fb9ad8f73bc0c | q_4f3fb9ad8f73bc0c | SELECT                          | false  | 0     |
+          |                    |                    |   "public"."cats"."breed"       |        |       |
+          |                    |                    | FROM                            |        |       |
+          |                    |                    |   "public"."cats"               |        |       |
+          |                    |                    | WHERE                           |        |       |
+          |                    |                    |   ("public"."cats"."name" = $1) |        |       |
+          +--------------------+--------------------+---------------------------------+--------+-------+
+        TABLE
```

Comment:
**_nitpick (100% skippable):_** This kind of reminds me of a feature spec (interactive, user-facing expectation). Non-specific, bare-minimum matchers could help with spec brittleness. Such as: "I expect these values to be in there somewhere" and "I expect these columns to be there somewhere". 


```
  scenario 'send shipment' do
    # note: `have_content` means *anywhere* on the webpage
    expect(page).to have_content('Ready to ship')

    within '.qa-passed' do
      click_button 'Add to shipment'
    end
    
    within '.factory-second' do
      click_button 'Add to shipment'
    end
    
    within '.shipping-list' do
      click_button 'Send shipment'
    end

    expect(page).to have_content("Robots #{robot1.id}, #{robot2.id} is shipped successfully.")
  end
  ```

Reply:
I like that idea! I think for now, I'll keep things as-is in the interest of time, but this is a prime candidate for refactoring in the future!